/*
 * This file was automatically generated by EvoSuite
 * Thu Dec 09 05:04:00 GMT 2021
 */

package org.apache.commons.lang3.concurrent;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.commons.lang3.concurrent.ConcurrentException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class ConcurrentException_ESTest extends ConcurrentException_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      ConcurrentException concurrentException0 = null;
      try {
        concurrentException0 = new ConcurrentException((Throwable) null);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Not a checked exception: null
         //
         verifyException("org.apache.commons.lang3.concurrent.ConcurrentUtils", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      ConcurrentException concurrentException0 = null;
      try {
        concurrentException0 = new ConcurrentException("", (Throwable) null);
        fail("Expecting exception: IllegalArgumentException");
      
      } catch(IllegalArgumentException e) {
         //
         // Not a checked exception: null
         //
         verifyException("org.apache.commons.lang3.concurrent.ConcurrentUtils", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      ConcurrentException concurrentException0 = new ConcurrentException();
      ConcurrentException concurrentException1 = new ConcurrentException("org.apache.commons.lang3.concurrent.ConcurrentException", concurrentException0);
      assertFalse(concurrentException1.equals((Object)concurrentException0));
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      ConcurrentException concurrentException0 = new ConcurrentException();
      ConcurrentException concurrentException1 = new ConcurrentException(concurrentException0);
      assertFalse(concurrentException1.equals((Object)concurrentException0));
  }
}
